### ISSUE TITLE
the combine cascade function returns info instead of obj

### ISSUE THREAD

**Issue by ATGardner** [winstonjs/logform#12]
I am not sure, but I thought each formatter return a new info object (if I want to keep the original object immutable), and that object will be passed on to the next formatter (as it currently is), and returned as the final value of the cascade function (which it currently isn't).
Am I missing something?

**Comment by baer** [comment]
I'm not sure I 100% understand the question but I _can_ say that info objects are mutable. The following line is from the readme and if you look at any of the built-in formatters, you'll see that they set properties directly onto the object. 

> "Formats are prototypal objects (i.e. class instances) that define a single method: transform(info, opts) and return the mutated info"

**Comment by iamkirkbater** [comment]
The info object itself is mutable, but it also seems to be passed by reference?  In my case I'm trying to re-order all of my log messages and re-add the metadata object format that was used in Winston 2.x.

So I have an 'orderedFormatter` which takes the info argument, and tries to return a new object instead of the info object, with the parameters ordered in the way I would like them ordered:

```javascript
const orderedFormatter = winston.format((info) => {
  const newLogMessage = {
    level: info.level,
    message: info.message,
    metadata: {},
    timestamp: info.timestamp,
  };

  delete info.level;
  delete info.message;
  delete info.timestamp;
  newLogMessage.metadata = Object.assign({}, info);

  return newLogMessage;
});
```

When I run my logger, all that gets logged is `undefined`.  I can replace anything within the `info` object, but if I try to return anything other than the info object I get `undefined`.

If I don't return anything, the logger fails silently.

I'm going to try to figure out how to get this working, but I wanted to describe the functionality I'm currently seeing for reference.

**Comment by iamkirkbater** [comment]
Ha, yeah, it seems that just writing my thoughts down sparked the idea.  This seems to work for me:

```javascript
const orderedFormatter = winston.format((info) => {
  const level = info.level;
  const message = info.message;
  const timestamp = info.timestamp;
  const metadata = Object.assign({}, info);

  Object.keys(info).forEach((k) => {
    delete info[k];
  });

  info.level = level;
  info.message = message;
  info.metadata = metadata;
  info.timestamp = timestamp;

  return info;
});
```

If I get some time later I'll abstract out the two main things I'm doing here (restoring a metadata object and creating an ordering of the log message) and contribute them back to this repo, but the code in this post should allow you to manipulate your log messages.

**Comment by indexzero** [comment]
Going to close this as fixed by #14 based on the discussion from @iamkirkbater
